src/Common/Core/Impl/Shell/IIdleTimeService.cs

```diff
@@ -1,11 +1,14 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+using System;
 
-using System.Threading;
-
-namespace Microsoft.Languages.Editor.Host {
+namespace Microsoft.Common.Core.Shell {
     public interface IIdleTimeService {
-        void DoIdle();
+        /// <summary>
+        /// Fires when host application enters idle state.
+        /// </summary>
+        event EventHandler<EventArgs> Idle;
```

Comment:
This will be in x-plat. Activities related to parsing, task list, squiggles need idle for UI thread activities.